core/llm.go

```diff
@@ -284,6 +305,18 @@ func (r *LLMRouter) LoadConfig(ctx context.Context, getenv func(context.Context,
 	if err != nil {
 		return err
 	}
+
+	historyReplay, err := getenv(ctx, "LLM_HISTORY_REPLAY")
```

Comment:
🤔 does this force us to only do dagger-in-dagger style integration tests instead of being able to call the API directly when appropriate? 

or is there some way to do `c := connect(ctx, t).WithEnv(LLM_HISTORY_REPLAY=%x)`?

Reply:
Sigh, yeah. We can only call the API nested.

I like the `WithEnv` idea a lot, but it's not really plumbed in, and it's a bit tricky, since `connect` doesn't *always* spawn a new subprocess with an env, e.g. in our tests where we're doing nested dagger.

:thinking: maybe we could put it in the model name? Something like `replay/<base64-encoded-replay>`? Or another parameter there? I don't really mind, the logic is pretty similar either way.

Reply:
> Something like replay/<base64-encoded-replay>? Or another parameter there? 

i feel like base64ing these replays could get waaaay too large, especially with the automatic recording... any tool call that returns a big text blob is gonna blow that field up real real big. maybe size isn't a real problem though, idk. an additional parameter sounds nice, though.

Reply:
Agreed, spun off a discord convo, since this affects our public facing API: https://discord.com/channels/707636530424053791/1351531596288753716/1351531917807321149